CONTRIBUTING.md

```diff
@@ -0,0 +1,90 @@
+
+To create a new issue, comment on an existing issue, or assign an issue to yourself, you'll need to first [create a JIRA account here](https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/registrations?client_id=jbossorg&response_type=code&scope=openid&redirect_uri=https%3A%2F%2Fsso.jboss.org%2Flogin%3Fservice%3Dhttps%253A%252F%252Fissues.redhat.com%26ffp%3DRedHatExternalProvider).
+
+
+### Good First Issues
+Want to contribute to the WildFly Elytron project but aren't quite sure where to start? Check out our issues with the `good-first-issue` label. These are a triaged set of issues that are great for getting started on our project. [These can be found here](https://issues.redhat.com/issues/?filter=12364234). 
+
+Once you have selected an issue you'd like to work on, make sure it's not already assigned to someone else. Then, remember to assign it to yourself to prevent someone else from also working on the same issue.
+
+![jira](https://i.imgur.com/uEz7v32.png)
+
+## Setup Development Environment
+You will need:
+
+* JDK 11
+* Git
+* Maven 3.3.9 or later
+* An [IDE](https://en.wikipedia.org/wiki/Comparison_of_integrated_development_environments#Java)
+(e.g., [IntelliJ IDEA](https://www.jetbrains.com/idea/download/), [Eclipse](https://www.eclipse.org/downloads/), etc.)
+
+First `cd` to the appropriate directory (eg: `cd wildfly-elytron`)
+
+Add a remote ref to upstream, for pulling future updates.
+For example:
+
+```
+git remote add upstream https://github.com/wildfly-security/wildfly-elytron
+```
+To build `wildfly-elytron` run:
+```bash
+mvn clean install
+```
+
+To skip the tests, use:
+
+```bash
+mvn clean install -DskipTests=true
+```
+
+To run only a specific test, use:
+
+```bash
+mvn clean install -Dtest=TestClassName
+```
+For more information, details, or integration with WildFly and WildFly Core, visit https://wildfly-security.github.io/wildfly-elytron/getting-started-for-developers/
+
+## Coding Guidlines
+
+When submitting a PR, please keep the following guidelines in mind:
+
+1. It is recommend that you use a separate branch for every Pull Request to submit. To keep things straightforward and memorable, you should try to name the branch the JIRA issue number. Doing so also allows your to prevent conflicts in the future when working on other issues. For example, for ELY-2127, you should also name the branch ELY-2127.
+
+1. In general, it's good practice to squash all of your commits into a single commit. For larger changes, it's ok to have multiple meaningful commits. If you need help with squashing your commits, feel free to ask us how to do this on your pull request. We're more than happy to help!
+
+2. Please include the JIRA issue you worked on in the title of your pull request and in your commit message. For example, for ELY-2127, the PR title and commit message should be `[ELY-2127] Add client side TRACE logging after successful authentication`.
```

Comment:
s/for ELY-2127/for [ELY-2127](https://issues.redhat.com/browse/ELY-2127)

Reply:
Changed